src/daq_config_server/log.py

```diff
@@ -1,24 +1,80 @@
 import logging
+from typing import Literal, TextIO
 
+from graypy import GELFTCPHandler
+from pydantic import BaseModel, HttpUrl
 
-# See https://github.com/DiamondLightSource/daq-config-server/issues/73
-# for making the logging configurable
-def get_default_logger(name: str = __name__) -> logging.Logger:
-    logger = logging.getLogger(name)
-    logger.setLevel(logging.INFO)
+LogLevel = Literal["NOTSET", "DEBUG", "INFO", "WARNING", "ERROR", "CRITICAL"]
 
-    # Prevent adding handlers multiple times
-    if not logger.handlers:
-        console_handler = logging.StreamHandler()
-        formatter = logging.Formatter(
-            "%(asctime)s - %(levelname)s - %(message)s", datefmt="%Y-%m-%d %H:%M:%S"
-        )
-        console_handler.setFormatter(formatter)
-        logger.addHandler(console_handler)
 
-    return logger
+class GraylogConfig(BaseModel):
+    enabled: bool = False
+    level: LogLevel = "INFO"
+    url: HttpUrl = HttpUrl("http://localhost:5555")
 
 
-# For now use a basic console-writing logger. Integrate properly with kubernetes in the
-# future
-LOGGER = get_default_logger("daq-config-server")
+class LoggingConfig(BaseModel):
+    graylog: GraylogConfig = GraylogConfig()
+
+
+def set_up_stream_handler(
+    logger: logging.Logger, logging_config: LoggingConfig
+) -> logging.StreamHandler[TextIO]:
+    """Creates and configures StreamHandler, then attaches to logger.
+
+    Args:
+        logger: Logger to attach handler to
+        logging_config: LoggingConfig
+    """
+    stream_handler = logging.StreamHandler()
+    stream_handler.setLevel("DEBUG")
```

Comment:
```suggestion
    graylog: GraylogConfig = GraylogConfig()
    level: LogLevel = "DEBUG"


def set_up_stream_handler(
    logger: logging.Logger, logging_config: LoggingConfig
) -> logging.StreamHandler[TextIO]:
    """Creates and configures StreamHandler, then attaches to logger.

    Args:
        logger: Logger to attach handler to
        logging_config: LoggingConfig
    """
    stream_handler = logging.StreamHandler()
    stream_handler.setLevel(logging_config.level)
```

nit

Reply:
Having `level` on the LoggingConfig and on the GraylogConfig would be how I'd do it.